### ISSUE TITLE
Text rendering on linux build / release

### ISSUE THREAD

**Issue by szobov** [ahrm/sioyek#119]
Hi there,

I just tried to use a fresh build image for [here](https://github.com/ahrm/sioyek/releases/tag/v1.0.0) and also built from the master, but in both cases had such a rendering issue:
![Selection_221](https://user-images.githubusercontent.com/7851779/149988269-a923a327-9bce-4dca-b312-4ed44d6bebff.png)
Could you please help me to fix it?
My env:
Xubuntu 20.04.1

Thanks in advance!

**Comment by ahrm** [comment]
Yes sometimes this happens to me too but it is usually fixed when I scroll the page. I don't really know what the problem is but if I had to guess I would say it has something to do with the texture lookup in OpenGL.

**Comment by vqhuy** [comment]
I got exactly the same problem, even though my pdf viewers (on arch) had bad text rendering for pdfs, but sioyek is worse, it looks like the image above even if I scroll the page.

**Comment by ahrm** [comment]
Added a linear filtering option in baed2b35baf00814ca155d6e4c28b60d61083e1e which might help with the issue. Please add the following to your `prefs_user.config` and see if the issue persists:

```
linear_filter    1
```

**Comment by vqhuy** [comment]
Thanks for a prompt response (and an awesome work)!
Unfortunately, I don't think it solves the problem.
I attached here the screenshot, on the left is sioyek, and on the right is zathura (both are far from fine for my eyes), but I think sioyek is a bit worse (even with `linear_filter 1`)
![image](https://user-images.githubusercontent.com/5112602/151787047-adec1045-f4a3-4f20-b727-70c37746d3a2.png)

**Comment by ahrm** [comment]
I am not entirely sure what the issue is, but my guess is that somehow the display resolution reported by your OS is different from the actual screen resolution. I added a `display_resolution_scale` option in d7549b3b017d9c22f7248d3b559be312d53ceb9a to manually set resolution scale. You can add the following to your `prefs_user.config`  file and see if it makes a difference:

```
display_resolution_scale     1
```
if it didn't help maybe try with other values instead of 1, for example:
```
display_resolution_scale     2
```

**Comment by szobov** [comment]
Hi @ahrm 
Thanks for your advice!
Unfortunately, none of them helped.

Just to let you know, recently I switched to a new laptop (the same model) and install fresh ubuntu. I copied all configs and also the config of sioyek. On the previous laptop sioyek worked like a charm, but now I experience this issue.

**Comment by vqhuy** [comment]
Thanks @ahrm for your efforts! Unfortunately, same for me.
I guess what is missing is the subpixel rendering/anti-aliasing, because I only had this problem in sioyek, zathura (which use mupdf as backend) but google chrome works just fine.

**Comment by vqhuy** [comment]
(I could create another issue if needed) I am wondering if there is a way to read fontconfig settings and render accordingly?

**Comment by ahrm** [comment]
> (I could create another issue if needed) I am wondering if there is a way to read fontconfig settings and render accordingly?

I am not sure but I don't think our rendering has anything to do with fontconfig settings. Rendering is handled entirely by mupdf, we just display the results.

**Comment by vqhuy** [comment]
Thanks, I guess so. I did some test, and apparently mupdf uses its own rendering settings.
I will try something with it and see if there is anything we can use here.
Thanks again for your awesome work!

**Comment by ahrm** [comment]
I have fixed an issue with rendering when multiple monitors with different resolutions are present in c446f7cf38830a7ac18ec5eb6ac2b30878fccd8b, I am not sure fixed this issue though.

**Comment by ahrm** [comment]
Yes, that is my guess too.

**Comment by ahrm** [comment]
First of all thanks for your investigation, this is very helpful.
I am not sure what is the best way to do it. I will have to try out some possible solutions next week.

**Comment by ahrm** [comment]
I think this issue at long last should be fixed in 8c2b4565b575c3c153d26da2492f5f396ca46621 thanks to @lennonhill 's investigation.

**Comment by nomppy** [comment]
I think I'm running into this issue again. Here are two screenshots of the same pdf opened in Sioyek, on monitors with different resolutions. I am on the latest version (2.0.0) of Sioyek, installed from homebrew cask on macos.

![image](https://github.com/user-attachments/assets/cc13fcd0-6ee9-4243-ab1b-da4bd1047723)
<img width="968" alt="image" src="https://github.com/user-attachments/assets/d5c38535-3a1a-4554-af8d-efccf86eca43">

**Comment by ahrm** [comment]
Does it happen on the experimental build of the release branch here: https://github.com/ahrm/sioyek/releases/tag/sioyek3-alpha0?
